core/templates/pages/contributor-dashboard-page/services/contribution-and-review-stats.service.ts

```diff
@@ -0,0 +1,131 @@
+
+/**
+ * @fileoverview Service for fetching and resolving contribution
+ * and review stats.
+ */
+
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+import { ContributionAndReviewStatsBackendApiService }
+  from './contribution-and-review-stats-backend-api.service';
+
+export interface TranslationContributionBackendDict {
+   'language_code': string;
+   'topic_name': string;
+   'submitted_translations_count': number;
+   'submitted_translation_word_count': number;
+   'accepted_translations_count': number;
+   'accepted_translations_without_reviewer_edits_count': number;
+   'accepted_translation_word_count': number;
+   'rejected_translations_count': number;
+   'rejected_translation_word_count': number;
+   'first_contribution_date': string;
+   'last_contribution_date': string;
+}
+
+export interface TranslationReviewBackendDict {
+   'language_code': string;
+   'topic_name': string;
+   'reviewed_translations_count': number;
+   'reviewed_translation_word_count': number;
+   'accepted_translations_count': number;
+   'accepted_translations_with_reviewer_edits_count': number;
+   'accepted_translation_word_count': number;
+   'first_contribution_date': string;
+   'last_contribution_date': string;
+}
+
+export interface QuestionContributionBackendDict {
+   'topic_name': string;
+   'submitted_questions_count': number;
+   'accepted_questions_count': number;
+   'accepted_questions_without_reviewer_edits_count': number;
+   'first_contribution_date': string;
+   'last_contribution_date': string;
+}
+
+export interface QuestionReviewBackendDict {
+   'topic_name': string;
+   'reviewed_questions_count': number;
+   'accepted_questions_count': number;
+   'accepted_questions_with_reviewer_edits_count': number;
+   'first_contribution_date': string;
+   'last_contribution_date': string;
+}
+
+export interface ContributorStatsSummaryBackendDict {
+   'translation_contribution_stats': [TranslationContributionBackendDict];
+   'translation_review_stats': [TranslationReviewBackendDict];
+   'question_contribution_stats': [QuestionContributionBackendDict];
+   'question_review_stats': [QuestionReviewBackendDict];
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ContributionAndReviewStatsService {
+  constructor(
+    private contributionAndReviewStatsBackendApiService:
+      ContributionAndReviewStatsBackendApiService
+  ) {}
+
+  async fetchAllStats(
+      username: string
+  ): Promise<ContributorStatsSummaryBackendDict> {
+    return (
+      this.contributionAndReviewStatsBackendApiService
+        .fetchAllContributionAndReviewStatsAsync(username));
+  }
+
+  async fetchTranslationContributionStats(
+      username: string
+  ): Promise<ContributorStatsSummaryBackendDict> {
+    return (
+      this.contributionAndReviewStatsBackendApiService
+        .fetchContributionAndReviewStatsAsync(
+          'translation', 'submission', username));
```

Comment:
These hardcoded strings ought to be named constants / enums, I think.

Also these are four roundtrip calls. How long does each take? I'm wondering about this strategy vs batching the GET into a single one.

Reply:
> These hardcoded strings ought to be named constants / enums, I think.

Done

> How long does each take?

```
Translation Submissions -> 0.0019922256469726562 s
Translation Reviews -> 0.002099275588989258 s
Question Submissions -> 0.0017049312591552734 s
Question Reviews -> 0.0022652149200439453 s
------------------------------------------------------
Total -> 0.00806164741 s
```

Also I have a separate endpoint fetch all stats at once in case to fetch everything from one request. It takes 0.008791923522949219 s.

Reply:
Just to check, these are being measured starting from the frontend? I'm a bit surprised because I'd have thought the network latency would be longer than that.